web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -46,8 +78,76 @@ describe("<SidebarCardFundingNudge />", () => {
     card = generateSidebarCardContent({ id: fundingNudge });
   });
 
+  const generateTaxFilingUserData = (params: {
+    publicFiling: boolean;
+    formedInNavigator?: boolean;
+    businessName?: string;
+    responsibleOwnerName?: string;
+    taxId?: string;
+  }): UserData => {
+    let formationData: FormationData = {
+      formationFormData: createEmptyFormationFormData(),
+      formationResponse: undefined,
+      getFilingResponse: undefined,
+      completedFilingPayment: false,
+      businessNameAvailability: undefined,
+      dbaBusinessNameAvailability: undefined,
+      lastVisitedPageIndex: 0,
+    };
+
+    const legalStructureId = params.publicFiling
+      ? randomPublicFilingLegalType()
+      : randomLegalStructure({ requiresPublicFiling: false }).id;
+    if (params.publicFiling) {
+      formationData = generateFormationData(
+        {
+          completedFilingPayment: !!params.formedInNavigator,
+          getFilingResponse: generateGetFilingResponse({ success: params.formedInNavigator }),
+        },
+        legalStructureId as FormationLegalType
+      );
+    }
+
+    return generateUserDataForBusiness(
+      generateBusiness({
+        profileData: generateProfileData({
+          legalStructureId: legalStructureId,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayTaxAccessButton;
+            })
+          ).id,
+          taxId: params.taxId ? `*${params.taxId.slice(1)}` : "",
+          encryptedTaxId: params.taxId ? `encrypted-${params.taxId}` : "",
+          businessName: params.businessName || "",
+          responsibleOwnerName: params.responsibleOwnerName || "",
+        }),
+        formationData: formationData,
+        taxFilingData: generateTaxFilingData({ state: undefined }),
+      })
+    );
+  };
+
   describe("when clicking funding button for non-generic industry", () => {
+    const userDataWithPrefilledFields = generateTaxFilingUserData({
+      publicFiling: true,
+      taxId: "123456789123",
+      businessName: "MrFakesHotDogBonanza",
```

Comment:
[sand] If we're using the same setup every time, I don't know that it necessarily makes sense for us to pass these as props.

Reply:
Thinking specifically because this function is only used in one place.